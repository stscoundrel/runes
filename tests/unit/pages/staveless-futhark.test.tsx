import ReactDOM from 'react-dom'
import StavelessFuthark from 'pages/staveless-futhark'
import renderer from 'react-test-renderer'

describe('Staveless Futhark runic keyboard page', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<StavelessFuthark />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<StavelessFuthark />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
