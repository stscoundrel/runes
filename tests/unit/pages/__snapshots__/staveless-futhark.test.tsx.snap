// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Staveless Futhark runic keyboard page Matches snapshot 1`] = `
Array [
  <main
    className="container"
  >
    <section>
      <header>
        <h1>
          Younger Futhark
        </h1>
        <p>
          Translate runes to text
        </p>
      </header>
      <p
        className="result"
      />
      <div
        className="keyboard"
      >
        <button
          aria-label="Runic key ᛙ, meaning letter f"
          className="key"
          onClick={[Function]}
        >
          ᛙ
        </button>
        <button
          aria-label="Runic key ╮, meaning letter u"
          className="key"
          onClick={[Function]}
        >
          ╮
        </button>
        <button
          aria-label="Runic key ו, meaning letter þ"
          className="key"
          onClick={[Function]}
        >
          ו
        </button>
        <button
          aria-label="Runic key ˎ, meaning letter o"
          className="key"
          onClick={[Function]}
        >
          ˎ
        </button>
        <button
          aria-label="Runic key ◟, meaning letter r"
          className="key"
          onClick={[Function]}
        >
          ◟
        </button>
        <button
          aria-label="Runic key ᛍ, meaning letter k"
          className="key"
          onClick={[Function]}
        >
          ᛍ
        </button>
        <button
          aria-label="Runic key ᚽ, meaning letter h"
          className="key"
          onClick={[Function]}
        >
          ᚽ
        </button>
        <button
          aria-label="Runic key ⸜, meaning letter n"
          className="key"
          onClick={[Function]}
        >
          ⸜
        </button>
        <button
          aria-label="Runic key ᛁ, meaning letter i"
          className="key"
          onClick={[Function]}
        >
          ᛁ
        </button>
        <button
          aria-label="Runic key ⸝, meaning letter a"
          className="key"
          onClick={[Function]}
        >
          ⸝
        </button>
        <button
          aria-label="Runic key ╵, meaning letter s"
          className="key"
          onClick={[Function]}
        >
          ╵
        </button>
        <button
          aria-label="Runic key ⸍, meaning letter t"
          className="key"
          onClick={[Function]}
        >
          ⸍
        </button>
        <button
          aria-label="Runic key ˏ, meaning letter b"
          className="key"
          onClick={[Function]}
        >
          ˏ
        </button>
        <button
          aria-label="Runic key ⠃, meaning letter m"
          className="key"
          onClick={[Function]}
        >
          ⠃
        </button>
        <button
          aria-label="Runic key ⸌, meaning letter l"
          className="key"
          onClick={[Function]}
        >
          ⸌
        </button>
        <button
          aria-label="Runic key ⡄, meaning letter R"
          className="key"
          onClick={[Function]}
        >
          ⡄
        </button>
        <button
          aria-label="Runic key :, meaning letter  "
          className="key"
          onClick={[Function]}
        >
          :
        </button>
      </div>
    </section>
  </main>,
  <footer
    className="section"
  >
    <div
      className="container text-center"
    >
      <small
        className="copyright"
      >
        Copyright © 
        2021
        <a
          href="https://github.com/stscoundrel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sampo Silvennoinen / StScoundrel
        </a>
      </small>
    </div>
  </footer>,
]
`;
